src/gui/instrument/InstrumentTrackWindow.cpp

```diff
@@ -717,5 +698,66 @@ void InstrumentTrackWindow::adjustTabSize(QWidget *w)
 	w->update();
 }
 
+QMdiSubWindow* InstrumentTrackWindow::findSubWindowInParents()
+{
+	// TODO Move to helper? Does not seem to be provided by Qt.
+	auto p = parentWidget();
+
+	while (p != nullptr)
+	{
+		auto mdiSubWindow = dynamic_cast<QMdiSubWindow*>(p);
+		if (mdiSubWindow)
+		{
+			return mdiSubWindow;
+		}
+		else
+		{
+			p = p->parentWidget();
+		}
+	}
+
+	return nullptr;
+}
+
+void InstrumentTrackWindow::updateSubWindowState()
```

Comment:
What I find strange about this method: It is no way specific to Instrument Track SubWindows (except the "isResizable" which I plan to generalize to both Instruments and Effects, i.e. to `PluginView`, but not all `SubWindow` types - nonetheless, `SubWindow` might just do a `qobject_cast` on its children, and if it's a PluginView, use the `isResizable` from there).

I guess your PR is only required for instruments, because they are the only SubWindow type that can be maximized and can both be resizable or not? In this case, the issue will occur for Effects sooner or later. And for the other SubWindow types, I assume it might not harm?

The advantage of putting that code into `SubWindow` would be cleaner encapsulation and no need to find the parent.

Reply:
Resolving this conversation as it was moved out into the main thread with this comment: https://github.com/LMMS/lmms/pull/7514#issuecomment-2406838814